src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,848 @@
+            else :
+                cephfs.mkdir(path, permission)
+        except:
+             traceback.print_exc(file = sys.stdout)
+
+    def complete_put(self, text, line, begidx, endidx):
+        """ 
+        auto complete of file name.
+        """    
+        index_dict = {1: self.path_complete}
+        return self.index_based_complete(text, line, begidx, endidx, index_dict)
+
+    put_parser = argparse.ArgumentParser(description = 'Copy a file to Ceph File System from Local Directory.')
+    put_parser.add_argument('local_path', type = str, help = 'Path of the file in the local system')
+    put_parser.add_argument('remote_path', type = str, help = 'Path of the file in the remote system')
+    put_parser.add_argument('-f', '--force', action = 'store_true', help = 'Overwrites the destination if it already exists.')
+
+    @with_argparser(put_parser)
+    def do_put(self, args):
+        """ 
+        Copy a file to Ceph File System from Local Directory.
+            Usage : put <local_path> <remote_path> 
+        """
+        try:       
+            root_src_dir = args.local_path 
+            root_dst_dir = args.remote_path 
+            if args.local_path == '.':
+                root_src_dir = os.getcwd()
+            if args.remote_path == '.':
+                root_dst_dir = cephfs.getcwd().decode('utf-8') 
+            elif root_dst_dir[-1] != '/':
+                root_dst_dir += '/'
+            if args.local_path == '-'or os.path.isfile(root_src_dir):
+                copy_from_local(root_src_dir, root_dst_dir)
+            else :
+                for src_dir, dirs, files in os.walk(root_src_dir):
+                    dst_dir = src_dir.replace(root_src_dir, root_dst_dir, 1)
+                    if not args.force and dst_dir != '/':
+                        cephfs.mkdirs(to_bytes(dst_dir), 0o777)
+                    elif args.force and not is_dir_exists(dst_dir):
+                        cephfs.mkdirs(to_bytes(dst_dir), 0o777)
+                    for file_ in files:
+                        src_file = os.path.join(src_dir, file_)
+                        dst_file = '/'+ dst_dir + '/'+ file_
+                        if args.force and is_file_exists(dst_file):
+                            cephfs.unlink(to_bytes(dst_file))
+                        copy_from_local(src_file, dst_file.replace('//', '/'))
+        except :
+             traceback.print_exc(file = sys.stdout)
+    
+    def complete_get(self, text, line, begidx, endidx):
+        """ 
+        auto complete of file name.
+        """    
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    get_parser = argparse.ArgumentParser(description = 'Copy a file from Ceph File System from Local Directory.')
+    get_parser.add_argument('remote_path', type = str, help = 'Path of the file in the remote system')
+    get_parser.add_argument('local_path', type = str, help = 'Path of the file in the local system')
```

Comment:
`local_path` should be optional. The default should be `os.path.basename(remote_path)`.